server/backends/chunkstore/chunkstore.go

```diff
@@ -399,8 +399,13 @@ type ChunkstoreWriter struct {
 }
 
 func (w *ChunkstoreWriter) readFromWriteResultChannel() (int, error) {
+	delay := time.NewTimer(w.writeTimeoutDuration)
 	select {
 	case result, open := <-w.writeResultChannel:
+		if !delay.Stop() {
+			// don't leak the timer
+			<-delay.C
+		}
```

Comment:
nit: it would be slightly more robust IMO to have this in a `defer` just after declaring `delay` - same in the diff below

Reply:
sgtm, I'll do that

Reply:
mmm, doesn't work because we already drained the channel in the `<-delay.C` case, so it deadlocks in the `defer`.

Reply:
solved it; we can read from the channel only if a value is ready in the defer.

Reply:
Fixed.